File: backend/models/payment.go
package models

import (
	"time"

	"gorm.io/gorm"
)

type Payment struct {
	ID              uint           `gorm:"primaryKey" json:"id"`
	TenantID        uint           `json:"tenant_id"`
	Tenant          Tenant         `gorm:"foreignKey:TenantID" json:"tenant"`
	Amount          float64        `json:"amount"`
	DueDate         time.Time      `json:"due_date"`         // Tanggal jatuh tempo
	OverdueDate     time.Time      `json:"overdue_date"`     // Max overdue (DueDate + 10 hari)
	BillingPeriod   string         `gorm:"size:50" json:"billing_period"` // e.g. "Juni 2026"
	Status          string         `gorm:"size:20" json:"status"`  // pending, paid, overdue, cancelled
	MidtransOrderID string         `gorm:"size:100" json:"midtrans_order_id"`
	PaidAt          *time.Time     `json:"paid_at"`          // Waktu pembayaran dikonfirmasi Midtrans
	PaymentDate     time.Time      `json:"payment_date"`     // Legacy: kept for backward compat
	Phone           string         `gorm:"-" json:"phone,omitempty"`
	CreatedAt       time.Time      `json:"created_at"`
	UpdatedAt       time.Time      `json:"updated_at"`
	DeletedAt       gorm.DeletedAt `gorm:"index" json:"-"`
}

File: rumaowner/lib/models/payment.dart
import 'tenant.dart';

class Payment {
  final int id;
  final int tenantId;
  final double amount;
  final DateTime dueDate;
  final String billingPeriod;
  final DateTime paymentDate;
  final DateTime? paidAt;
  final String status;
  final DateTime createdAt;
  final Tenant? tenant;

  Payment({
    required this.id,
    required this.tenantId,
    required this.amount,
    required this.dueDate,
    this.billingPeriod = '',
    required this.paymentDate,
    this.paidAt,
    required this.status,
    required this.createdAt,
    this.tenant,
  });

  factory Payment.fromJson(Map<String, dynamic> json) {
    return Payment(
      id: json['id'] ?? 0,
      tenantId: json['tenant_id'] ?? 0,
      amount: (json['amount'] as num?)?.toDouble() ?? 0.0,
      dueDate: json['due_date'] != null
          ? DateTime.parse(json['due_date'])
          : DateTime.now(),
      billingPeriod: json['billing_period'] ?? '',
      paymentDate: json['payment_date'] != null
          ? DateTime.parse(json['payment_date'])
          : DateTime.now(),
      paidAt: json['paid_at'] != null ? DateTime.parse(json['paid_at']) : null,
      status: json['status'] ?? 'pending',
      createdAt: json['created_at'] != null
          ? DateTime.parse(json['created_at'])
          : DateTime.now(),
      tenant: json['tenant'] != null ? Tenant.fromJson(json['tenant']) : null,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'tenant_id': tenantId,
      'amount': amount,
      'due_date': dueDate.toIso8601String(),
      'billing_period': billingPeriod,
      'payment_date': paymentDate.toIso8601String(),
      'paid_at': paidAt?.toIso8601String(),
      'status': status,
      'created_at': createdAt.toIso8601String(),
    };
  }
}
